From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Igor Pecovnik <igor.pecovnik@gmail.com>
Date: Fri, 3 Jan 2020 11:37:00 +0100
Subject: [ARCHEOLOGY] Move rockchip64 legacy to most recent kernel. Tested
 with standard https://github.com/armbian/autotests stress test on RockPro64
 and Rock64.

> X-Git-Archeology: - Revision 42ecf1582f83241e18ce329193c5eac66fd244de: https://github.com/armbian/build/commit/42ecf1582f83241e18ce329193c5eac66fd244de
> X-Git-Archeology:   Date: Fri, 03 Jan 2020 11:37:00 +0100
> X-Git-Archeology:   From: Igor Pecovnik <igor.pecovnik@gmail.com>
> X-Git-Archeology:   Subject: Move rockchip64 legacy to most recent kernel. Tested with standard https://github.com/armbian/autotests stress test on RockPro64 and Rock64.
> X-Git-Archeology:
> X-Git-Archeology: - Revision 5597ecfe073a70e11a3d832ca210471217706a0f: https://github.com/armbian/build/commit/5597ecfe073a70e11a3d832ca210471217706a0f
> X-Git-Archeology:   Date: Fri, 03 Jan 2020 11:42:56 +0100
> X-Git-Archeology:   From: Igor Pecovnik <igorpecovnik@users.noreply.github.com>
> X-Git-Archeology:   Subject: Move rockchip64 legacy to most recent kernel. Tested with standard https://github.com/armbian/autotests stress test on RockPro64 and Rock64. (#1714)
> X-Git-Archeology:
> X-Git-Archeology: - Revision 17bb3b5075613e344faedc4093eee5ea145f1b2e: https://github.com/armbian/build/commit/17bb3b5075613e344faedc4093eee5ea145f1b2e
> X-Git-Archeology:   Date: Sat, 04 Jan 2020 21:07:25 +0100
> X-Git-Archeology:   From: Igor Pecovnik <igorpecovnik@users.noreply.github.com>
> X-Git-Archeology:   Subject: Update RK3399 legacy to latest upstream kernel version (#1718)
> X-Git-Archeology:
> X-Git-Archeology: - Revision 0cdffb29b07305209efb12cf3b5ac6032d3a1153: https://github.com/armbian/build/commit/0cdffb29b07305209efb12cf3b5ac6032d3a1153
> X-Git-Archeology:   Date: Wed, 24 Mar 2021 19:01:53 +0100
> X-Git-Archeology:   From: Igor Pecovnik <igorpecovnik@users.noreply.github.com>
> X-Git-Archeology:   Subject: Renaming DEV branch to EDGE (#2704)
> X-Git-Archeology:
---
 Makefile        | 2 +-
 net/core/sock.c | 6 ++----
 2 files changed, 3 insertions(+), 5 deletions(-)

diff --git a/Makefile b/Makefile
index 1843aa8b69e9..526311d417a7 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 VERSION = 4
 PATCHLEVEL = 4
-SUBLEVEL = 204
+SUBLEVEL = 205
 EXTRAVERSION =
 NAME = Blurry Fish Butt
 
diff --git a/net/core/sock.c b/net/core/sock.c
index 72f54ca95ccd..8f2f5d497dc7 100644
--- a/net/core/sock.c
+++ b/net/core/sock.c
@@ -951,12 +951,10 @@ set_rcvbuf:
 			clear_bit(SOCK_PASSSEC, &sock->flags);
 		break;
 	case SO_MARK:
-		if (!ns_capable(sock_net(sk)->user_ns, CAP_NET_ADMIN)) {
+		if (!ns_capable(sock_net(sk)->user_ns, CAP_NET_ADMIN))
 			ret = -EPERM;
-		} else if (val != sk->sk_mark) {
+		else
 			sk->sk_mark = val;
-			sk_dst_reset(sk);
-		}
 		break;
 
 	case SO_RXQ_OVFL:
-- 
Armbian

